pull.go

```diff
@@ -179,6 +180,12 @@ func pullImage(ctx context.Context, store storage.Store, imageName string, optio
 	case signature.PolicyRequirementError:
 		logrus.Debugf("PolicyRequirementError raised")
 		return nil, err
+	case *url.Error:
+		logrus.Debugf("URL Error raised, tranport: [%v]", options.Transport)
+		if options.Transport == "" {
+			// Not using localhost, a url error that's a problem
```

Comment:
- When is the `options.Transport == ""` check true? AFAICT it is always non-empty, both because `cmd/*` set it to `DefaultTransport`, and because `NewBuilder` does.

- I have no idea what “not using localhost” means. Is that `docker://localhost/foo`, `localhost/foo`, both, neither?

- It is not obvious to me at all that various kinds errors should be handled differently in here (as I argue in #907 , most of that should probably be in `resolveImage`, along with the search loop), and similarly to #907, the `options.Transport == ""` check seems like an imprecise way to determine whether the registry search list is involved. `utils.ResolveName` knows precisely whether that happened, no need to heuristically guess again.

(I would be quite fine with this PR without the `options.Transport == ""` check, whether or not the code returns original `err` here.)

Reply:
The problem is if you do `buildah from alpine` the pull process first tries to pull it from localhost/alpine before going out to the registries.   This call using localhost/alpine returns a url error if a localhost registry isn't set up.   Prior to all these changes we just ignored the error from the localhost lookup  and then tried looking for alpine based on the entries in registries.conf.  Now it fails on that localhost lookup.  Dan found that if the transport isn't specified, then we know someone has given us just 'alpine' and we can then continue on and ignore the url error.

The thing that's biting me with this change is in some cases it's OK to get a failure and others not.  Trying to specify which are good and which are bad errors has proven challenging especially as we're pushing to try and get a release out.

Reply:
- I can’t see that `options.Transport == ""` can happen (#907 makes it happen for the brief time necessary to set `useRegistries`; but that’s not included here); perhaps I am just being blind.

- Again, `utils.ResolveName` _knows precisely_ whether the input contained a registry or not.

- `options.Transport == ""` (as well as the “input contained a registry” indication potentially returned by `utils.ResolveName`) is constant for all the names returned by `utils.ResolveName`; neither actually answers the question “are we pulling from `localhost`”

- The error handling should be much easier to do in `resolveImage`.